engine/src/test/java/io/zeebe/engine/processing/streamprocessor/StreamProcessorReprocessingTest.java

```diff
@@ -87,6 +92,85 @@ public void shouldCallRecordProcessorLifecycle() {
     inOrder.verifyNoMoreInteractions();
   }
 
+  @Test
+  public void shouldNotSwitchToProcessingAfterReprocessingWhenPaused() throws Exception {
+    // given - bunch of events to reprocess
+    IntStream.range(0, 5000)
+        .forEach(i -> streamProcessorRule.writeWorkflowInstanceEvent(ELEMENT_ACTIVATING, i));
+    final long sourceEvent = streamProcessorRule.writeWorkflowInstanceEvent(ELEMENT_ACTIVATING, 1);
+    streamProcessorRule.writeWorkflowInstanceEventWithSource(
+        WorkflowInstanceIntent.ELEMENT_ACTIVATED, 1, sourceEvent);
+
+    Awaitility.await()
+        .until(
+            () ->
+                streamProcessorRule
+                    .events()
+                    .onlyWorkflowInstanceRecords()
+                    .withIntent(ELEMENT_ACTIVATED),
+            StreamWrapper::exists);
+
+    // when
+    final var countDownLatch = new CountDownLatch(1);
+    final var streamProcessor =
+        streamProcessorRule.startTypedStreamProcessor(
+            (processors, context) ->
+                processors.withListener(
+                    new StreamProcessorLifecycleAware() {
+                      @Override
+                      public void onRecovered(final ReadonlyProcessingContext context) {
+                        countDownLatch.countDown();
+                      }
+                    }));
+    streamProcessor.pauseProcessing();
+    final var success = countDownLatch.await(15, TimeUnit.SECONDS);
+
+    // then
+    assertThat(success).isTrue();
+    final var currentPhase = streamProcessor.getCurrentPhase().join();
+    assertThat(currentPhase).isEqualTo(Phase.PAUSED);
```

Comment:
I don't like to verify the behavior by exposing an internal state. Maybe, the `currentPhase` is set correctly but the processor behaves wrong. 

Can we test the behavior instead? Like in the other test, write a record and verify that it is not read by the stream processor.

Reply:
Then we have the same issue as above right long timeout? :sweat_smile:

Reply:
:see_no_evil: